src/util.cpp

```diff
@@ -107,7 +107,7 @@ std::atomic<uint32_t> logCategories(0);
 static std::unique_ptr<CCriticalSection[]> ppmutexOpenSSL;
 void locking_callback(int mode, int i, const char* file, int line) NO_THREAD_SAFETY_ANALYSIS
 {
-    if (mode & CRYPTO_LOCK) {
+    if (mode) {
```

Comment:
Does the removal of CRYPTO_LOCK here have any influence on older versions of OpenSSL which need this?
Might be better to add the define here, if it's not defined.